tutorials/groups.qmd

```diff
@@ -0,0 +1,244 @@
+
+## Relationship between groups, metrics and connections
+
+Group structure provides a canonical way to define [📖 exponential](https://en.wikipedia.org/wiki/Exponential_map_(Lie_theory)) and logarithmic maps from the Lie algebra.
+They can be calculated in `Manifolds.jl` using the [`exp_lie`](@ref) and [`log_lie`](@ref) functions. Such exponential and logarithmic maps can be extended invariantly to tangent spaces at any point of the Lie group. This extension is implemented using functions [`exp_inv`](@ref) and [`log_inv`](@ref).
+
+Finally, there are `log` and `exp` functions which are metric (or connection)-related functions in `Manifolds.jl`. For groups which can be equipped with a bi-invariant metric, `log` and `log_inv` return the same result, similarly `exp` and `exp_inv`. However, only compact groups and their products with Euclidean spaces can have a bi-invariant metric (see for example Theorem 21.9 in [GallierQuaintance:2020](@cite)). A prominent example of a Lie group without a bi-invariant metric is the special Euclidean group (in two or more dimensions). Then we have a choice between a metric but non-invariant exponential map (which is generally the default choice for `exp`) or a non-metric, invariant exponential map (`exp_inv`). Which one should be used depends on whether being metric or being invariant is more important in a particular application.
+
+```{julia}
+G = SpecialEuclidean(2)
+p = ArrayPartition([1.0, -1.0], xRi)
+X = ArrayPartition([2.0, -3.0], aXi)
+q_m = exp(G, p, X)
+println(q_m)
+q_i = exp_inv(G, p, X)
+println(q_i)
+
+```
+
+As we can see, the results differ. We can observe the invariance as follows:
+```{julia}
+p2 = ArrayPartition([2.0, -1.0], xRi)
+q1_m = exp(G, translate(G, p2, p), translate_diff(G, p2, p, X))
+q2_m = translate(G, p2, exp(G, p, X))
+println(isapprox(q1_m, q2_m))
+
+q1_i = exp_inv(G, translate(G, p2, p), translate_diff(G, p2, p, X))
+q2_i = translate(G, p2, exp_inv(G, p, X))
+println(isapprox(q1_i, q2_i))
+```
+
+Now, `q1_m` and `q2_m` are different due to non-invariance of the metric connection but `q1_i` and `q2_i` are equal due to invarianced of `exp_inv`.
+
+The following table outlines invariance of `exp` and `log` of various groups.
+
+| Group | Zero torsion connection | Invariant |
+| --- | --- | --- |
+| `ProductGroup` | Product of connections in each submanifold | Yes if both connections are invariant separately, otherwise no |
+| `SemidirectProductGroup` | Same as underlying product | ❌ |
+| `TranslationGroup` | `CartanSchoutenZero` | ✅ |
+| `CircleGroup` | `CartanSchoutenZero` |✅|
+| `GeneralLinearGroup` | Metric connection from the left invariant metric induced from the standard basis on the Lie algebra | ❌ |
+| `GeneralUnitaryMultiplicationGroup` | `CartanSchoutenZero` (explicitly)  | ✅ |
+| `HeisenbergGroup` | `CartanSchoutenZero` | ✅ |
+| `SpecialLinearGroup` | Same as `GeneralLinear` | ❌ |
+
+: Invariance of `exp` and `log` for various groups {#tbl-exp-log-inv}
```

Comment:
@kellertuer do you know why this just renders as text "[TABLE]" in docs instead of the actual table? Do I need some  `{=commonmark}` thing around the table?

Reply:
I can check tomorrow (after my exams), I would have thought since this is in markdown that it stays as is, but maybe your Markdown is slightly off from the common mark quarto assumes? (To me it looks fine, but my idea of a markdown table is maybe also quite intuitive. Does it already `[TABLE]` in the markdown produced or only in Documenter? 

And sure, `{=commonmark}` would fix it probably but I would prefer fixing the actual error and not the symptom.

Reply:
"[TABLE]" is already in the .md file. I tried fixing it but nothing seems to work except `{=commonmark}`.

Reply:
I hope there is just a missing space somewhere, I can check later after my oral exams (probably around 5).

Reply:
Hm our setup is maybe broken on that even just an empty file with an example table from the quarto page renders into `[TABLE]` so I have to check what is happening there.

Reply:
🥳(after another oral exam) found it! Your lines were too long, but I could add a `variant:` to quarto, after reading about the corresponding Pandoc extension and now tables are rendered over as is if they are too long.

I also changed the column alignment to (left, left, center) instead of all of them being right aligned.

So this should now work, though the last column does not look that nice, maybe one could do the first with an orange circle or such and explain that below the table?

Reply:
Nice! But it looks like we still need `{=commonmark}`, right?

I will replace the first one with a circle.

Reply:
It works without, sorry I had that tested in an extra file to have that created faster and missed to adapt the code block. So I just committed that.
Makes typesetting tables easier then now :)

Reply:
Cool, thanks!